File: MCRRTImp/src/test/java/lab/mars/MCRRTImp/TestAttacker.java
package lab.mars.MCRRTImp;

import javafx.application.Application;
import javafx.scene.paint.Color;
import lab.mars.MCRRTImp.Vector2BasedImp.Attacker;
import lab.mars.MCRRTImp.model.Transform;
import lab.mars.MCRRTImp.infrastructure.ui.GUIBase;
import lab.mars.MCRRTImp.infrastructure.ui.Pencil;
import lab.mars.MCRRTImp.Vector2BasedImp.Vector2;
import org.apache.commons.math3.distribution.NormalDistribution;
import org.junit.Test;

import java.util.List;

public class TestAttacker {

    @Test
    public void testTimeScalar() {
        Attacker<Vector2> aircraft = new Attacker<>(new Vector2(5, 5), new Vector2(1, 1).normalize().scale(4.16666667), 10, 5, 200, 50, 2, null, null, null);
        double timeScalar = 1;
        while (timeScalar < 100) {
            List<Transform<Vector2>> transforms = aircraft.simulateKinetic(aircraft.velocity(), timeScalar);
            System.out.println("time scalar : " + timeScalar);
            System.out.println(transforms.get(0).position.angle(transforms.get(transforms.size() - 1).position));
            timeScalar ++;
        }
    }

    public static class TestTransform extends GUIBase {


        private Attacker<Vector2> aircraft = new Attacker<>(new Vector2(5, 5), new Vector2(1, 1).normalize().scale(4.16666667), 10, 5, 200, 50, 2, null, null, null);


        @Override
        protected void draw(Pencil pencil) {
            List<Transform<Vector2>> transforms = aircraft.simulateKinetic(aircraft.velocity(), 10);
            Vector2 origin = aircraft.position();
            double scaleBase = 1;
            pencil.scale(scaleBase).filled().color(Color.YELLOWGREEN).circle(origin, 1);
            for (Transform<Vector2> t : transforms) {
//                System.out.println(t.position);
                Vector2 position = t.position.cpy().translate(origin);
                Vector2 direction = t.velocity;
                Vector2 transformed = position.cpy().translate(direction);
                pencil.filled().color(Color.BLACK).circle(position, 5 / scaleBase);
                pencil.stroked(2).color(Color.BLACK).line(position, transformed);
            }
        }

        public static void main(String[] args) {
            Application.launch(args);
        }
    }
}

File: MCRRTImp/src/test/kotlin/lab/mars/MCRRTImp/Vector2BasedImp/TestAttacker.kt
package lab.mars.MCRRTImp.Vector2BasedImp

import javafx.application.Application
import javafx.scene.paint.Color
import infrastructure.Vector2BasedImp.Vector2
import infrastructure.model.Attacker
import lab.mars.MCRRTImp.model.Transform
import infrastructure.ui.GUIBase
import infrastructure.ui.Pencil
import org.junit.Test

class TestAttacker : GUIBase() {


        private val aircraft = Attacker(Vector2(100.0, 100.0), Vector2(1.0, 1.0).normalize().scale(4.16666667), 10.0, 100, 200.0, 50.0, 2.0, null, null, null)


        var deltaTime = 30.0

        override fun draw(pencil: Pencil) {
            val transforms = aircraft.simulateKinetic(aircraft.velocity, deltaTime)
            val origin = aircraft.position
            val scaleBase = 2.0
            pencil.scale(scaleBase).filled().color(Color.YELLOWGREEN).circle(origin, 1.0)
            for (t in transforms) {
                val position = t.position.cpy().scale(1.0).translate(origin)
                val direction = t.velocity
                val transformed = position.cpy().translate(direction)
                pencil.filled().color(Color.WHITE).circle(position, 5 / scaleBase)
                pencil.stroked(2.0).color(Color.WHITE).line(position, transformed)
            }
            deltaTime ++
            if (deltaTime > 60.0) {
                deltaTime = 1.0
            }
        }

    }



fun main(args: Array<String>) {
    Application.launch(TestAttacker::class.java, *args)
}
